tests/S-4_Account_creation _deletion_and _get_info.test.js

```diff
@@ -0,0 +1,369 @@
+test("Multipule usernames GET endpoint /user/{userName}/class/{className}", async (t) => {
+    const usernames = ["userName", "userName2", "userName3", "userName4", "userName5", "userName6"];
+    const className = "className"; // Don't change that because the dummy data returns only for this className
+
+    
+    // Pre-allocate the array based on the number of usernames
+    const responses = new Array(usernames.length);
+    
+    // Iterate over each username
+    for (let i = 0; i < usernames.length; i++) {
+        // Make the GET request
+        const response = await t.context.got.get(`user/${usernames[i]}/class/${className}`);
+        
+        // Check status Code
+        t.is(response.statusCode, 200, `Expected status code 200 for user: ${usernames[i]}`);
+        
+        // Assign the response to the pre-allocated array
+        responses[i] = response;
+    }
+    
+    const body = responses[0].body;
+    // Check that the first response has the expected body
+    
+    // Make sure it has a className and it is the expected one
+    t.not(body.className, undefined, "Response should have a className property");
+    t.is(body.className, className, `Expected className to be equal with that of the request ${className}`);
+
+    // Make sure it is an array and that it has a length
+    t.true(Array.isArray(body.users), "Users should be an array");
+    t.not(body.users.length,undefined, "Expected to be able to get lenght of users array");
+
+    // Loop over all users in the array
+    for (var i = 0 ; i < body.users.length ; i++ ){
+        const dummy_user = body.users[i];
+        // Check that the struct has the proper fields  
+        t.not(dummy_user.grade,          undefined, "User should have a grade property");
+        t.not(dummy_user.user,           undefined, "User should have a user property");
+        t.not(dummy_user.user.userName,  undefined, "User should have a userName property");
+        t.not(dummy_user.user.surname,   undefined, "User should have a surname property");
+        t.not(dummy_user.user.name,      undefined, "User should have a name property");
+        t.not(dummy_user.user.id,        undefined, "User should have a id property");
+        t.not(dummy_user.user.email,     undefined, "User should have a email property");        
+    }
+
+    // Check that all responses have the same body
+    for (let i = 1; i < responses.length; i++) {
+        t.deepEqual(
+            responses[i].body,
+            responses[0].body,
+            `Expected the same body response for ${usernames[i]} as for ${usernames[0]}`
+            );
+        }
+        
+});
+
+// Test for multipule usernames and classNames:    GET     /user/{userName}/class/{className}
+test("Multipule usernames and classNames GET endpoint /user/{userName}/class/{className}", async (t) => {
+    const usernames = ["userName1", "userName2", "userName3"];
+    const classNames = ["className1", "className2", "className3"];
```

Comment:
If we are going to have this test, do we need the previous ones?

Reply:
From a functionality point of view, not really, but the previous tests were created first because they were easier as a first test. 
All in all, they test a simpler case that I thought would be helpful for debugging if someone messes up the endpoint in the future.